message/messaging_service.cc

```diff
@@ -496,6 +503,8 @@ static constexpr unsigned do_get_rpc_client_idx(messaging_verb verb) {
     case messaging_verb::MUTATION_DONE:
     case messaging_verb::MUTATION_FAILED:
         return 3;
+    case messaging_verb::FORWARD_REQUEST:
+        return 4;
```

Comment:
Why do we need a new socket for this?

Reply:
To avoid the risk of deadlocks, details in [this Avi's comment](https://github.com/scylladb/scylla/pull/9209#discussion_r689645746).

Reply:
And why not reuse idx `3`?

Reply:
@avikivity is Gleb's suggestion ok for you?

Reply:
I think it works.

The two types of verbs are very different. MUTATION_DONE/MUTATION_FAILED are nowait messages. FORWARD_REQUEST is a round-trip request with high latency. Maybe we shouldn't mix them, causing write acknowledgements to be delayed behind these expensive requests.

Reply:
@avikivity Each socket takes resources. Adding one more index here increase number of sockets in the system proportional to number of shards. Latency of one request does not affect latency of the others on the same connection unless connection buffers are full which is not likely will be the case for aggregate requests that do most of the work on a nodes and send back only a single value. There is not problem to mix nowait and round-trip requests on the  same socket as well and we do it all the time.